src/items/items.service.ts

```diff
@@ -2,37 +2,67 @@ import { Injectable, NotFoundException } from '@nestjs/common';
 import { PrismaService } from 'src/prisma/prisma.service';
 import { CreateItemDto } from './dto/create-item.dto';
 import { UpdateItemDto } from './dto/update-item.dto';
-import { PaginationQueryDto } from './dto/pagination-query.dto';
+
 import { BadRequestException } from '@nestjs/common';
 import { InternalServerErrorException } from '@nestjs/common';
-
+import { Item } from '@prisma/client';
+import { ItemsRepository } from './items.repository';
+import { ItemsPaginationResponseDto } from './dto/items-pagination-response.dto';
+import { itemsPaginationQueryDto } from './dto/items-pagination-query.dto';
 @Injectable()
 export class ItemsService {
-  constructor(private readonly prisma: PrismaService) {}
+  constructor(
+    private readonly prisma: PrismaService,
+    private readonly itemsRepository: ItemsRepository,
+  ) {}
 
   //아이템 생성 createItem
   async createItem(createItemDto: CreateItemDto) {
-    try {
-      return await this.prisma.item.create({
-        data: createItemDto,
-      });
-    } catch (error) {
-      if (error.code === 'P2002') {
-        //P2002 : Prisma에러코드 : 유니크제약조건 위반 시 사용됨
-        throw new BadRequestException(
-          `이미 존재하는 아이템 이름입니다: ${createItemDto.name}`,
-        );
-      }
-      console.error('Database Error:', error); //서버 로그에 오류 기록
-      throw new InternalServerErrorException('서버 오류 발생'); //클라이언트에는 일반화된 오류 메시지 반환
+    const { name, mainCategoryId, subCategoryId } = createItemDto;
+
+    const item = await this.prisma.item.findUnique({
+      where: { name },
+    });
+    if (item) {
+      throw new BadRequestException(`이미 존재하는 아이템 이름입니다: ${name}`);
+    }
+
+    const mainCategory = await this.prisma.itemMainCategory.findUnique({
+      where: { id: mainCategoryId },
+    });
+    if (!mainCategory) {
+      throw new BadRequestException(
+        `존재하지 않는 메인 카테고리입니다: ${mainCategoryId}`,
+      );
     }
+
+    const subCategory = await this.prisma.itemSubCategory.findUnique({
+      where: { id: subCategoryId },
+    });
+    if (!subCategory) {
+      throw new BadRequestException(
+        `존재하지 않는 서브 카테고리입니다: ${subCategoryId}`,
+      );
+    }
+
+    return this.prisma.item.create({
+      data: createItemDto,
+    });
   }
   //아이템 전체 조회 getAllItems
-  async getAllItems(paginationQuery: PaginationQueryDto) {
-    const { limit = 8, offset = 0 } = paginationQuery;
-    return this.prisma.item.findMany({
-      skip: offset,
-      take: limit,
+  async getAllItems(
+    itemsPaginationQuery: itemsPaginationQueryDto,
+  ): Promise<ItemsPaginationResponseDto> {
+    const { limit = 8, page = 1 } = itemsPaginationQuery;
```

Comment:
```suggestion
    const { limit, page } = itemsPaginationQuery;
```
이미 컨트롤러에서 기본값을 설정해줬기 때문에 여기선 굳이 기본값을 넣어주지 않아도 될 것 같네요

Reply:
컨트롤러에서 dto 로 설정해줬다는 말씀인걸까요???😮